web/src/main/java/org/springframework/security/web/session/SessionInformationExpiredEvent.java

```diff
@@ -35,19 +36,22 @@ public final class SessionInformationExpiredEvent extends ApplicationEvent {
 
 	private final HttpServletResponse response;
 
+	private final FilterChain filterChain;
+
 	/**
 	 * Creates a new instance
 	 * @param sessionInformation the SessionInformation that is expired
 	 * @param request the HttpServletRequest
 	 * @param response the HttpServletResponse
 	 */
 	public SessionInformationExpiredEvent(SessionInformation sessionInformation, HttpServletRequest request,
-			HttpServletResponse response) {
+			HttpServletResponse response, FilterChain filterChain) {
```

Comment:
Instead of changing this constructor, a new constructor should be created accepting all those parameter + the `FilterChain`. The reason is because if we change this constructor it will be a breaking change for anyone that is using `SessionInformationExpiredEvent`.

Reply:
As you mentioned, I add new constructor below original constructor.  But filterChain value need for new constructor i had to handle this in original constructor too.